File: Medivic/src/app/model/remedio.ts
export class Remedio {
    idRemedio:number;
    nome: string;
    idUsuario: number;
    descricao: string;
    unidade: string;
    quantDias: number;
    intervalo: number;
    dosagem: number;
    dtInicio: string;
    horarioInicio: string;
    horarioNovo: string;
    horarios: string[];
    vezes: number;

    constructor(){
        this.idRemedio = 0;
        this.nome = "";
        this.idUsuario = 0;
        this.descricao = "";
        this.unidade = "";
        this.quantDias = 0;
        this.intervalo = 0;
        this.dosagem = 0;
        this.dtInicio = "";
        this.horarioInicio = "";
        this.horarioNovo = "";
        this.horarios = [];
        this.vezes = 0;
    }
}

File: medivicAPI/medivic/src/main/java/cefet/pp/medivic/model/Remedio.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cefet.pp.medivic.model;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

/**
 *
 * @author dougl
 */
@Setter
@Getter
@ToString
@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
public class Remedio {
    protected int idRemedio;
    protected int idUsuario;
    protected String nome;
    protected String descricao;
    protected String unidade;
    protected int quantDias;
    protected int intervalo;
    protected String dosagem;
    protected String dtInicio;
    protected String horarioNovo;
    protected String horarioInicio;
    protected String vezes;

    protected List<Horario> horarios;
}
